charmcraft/const.py

```diff
@@ -53,12 +55,26 @@
 # The minimum set of hooks to be provided for compatibility with old Juju
 MANDATORY_HOOK_NAMES = frozenset(("install", "start", "upgrade-charm"))
 
+CommonBaseStr = Literal[  # Bases supported as both build bases and run bases
+    "ubuntu@18.04",
+    "ubuntu@20.04",
+    "ubuntu@22.04",
+    "ubuntu@23.10",
+    "centos@7",
+    "almalinux@9",
+]
+BaseStr = CommonBaseStr | Literal["ubuntu@24.04"]
+BuildBaseStr = CommonBaseStr | Literal["ubuntu@devel"]
+
+DEVEL_BASE_STRINGS = ("ubuntu@24.04",)  # Bases that require a specified build base.
```

Comment:
the point was that from today we would not require build-base for 24.04

Reply:
new information makes my comment no longer valid

Reply:
Ohh right! Fixed :-)